### PR TITLE
60 if you sign out the basket doesnt get cleared

### PR DESCRIPTION
**Pull request by jcreek**
## Summary by CodeRabbit

- **New Features**
	- Enhanced sign-out process by integrating basket management, ensuring the basket is cleared upon user sign-out.

- **Bug Fixes**
	- Improved session handling by streamlining the redirection to the home page if no session is found.

- **Refactor**
	- Improved readability of the user update method in the account actions.
	- Removed the sign-out method from the account actions, indicating a change in available functionalities.

### CODE REVIEW

**Comment by jcreek** [comment]
@coderabbitai review

File: src/lib/components/SignOut.svelte [hunk 1]
```diff
@@ -9,10 +11,23 @@
 	// Access the supabase client from the layout data
 	export let supabase: any;
 
+	let localBasket: Basket;
+	const unsubscribe = basket.subscribe((value) => {
+		localBasket = value;
+	});
```
[the remark this answers is not in the document]
Comment (jcreek) [reply src/lib/components/SignOut.svelte:17]: Done

File: src/lib/components/SignOut.svelte [hunk 2]
```diff
@@ -9,10 +11,23 @@
...
 		const { error } = await supabase.auth.signOut().then(() => {
 			emitSignedOutEvent();
+
+			// Clear the basket
+			localBasket = { items: [], subtotal: 0 };
+			basket.set(localBasket);
+			unsubscribe();
+			if (typeof localStorage !== 'undefined') {
+				localStorage.removeItem('basket');
+			}
```
[the remark this answers is not in the document]
Comment (jcreek) [reply src/lib/components/SignOut.svelte:40]: Done

File: src/lib/components/SignOut.svelte [hunk 3]
```diff
@@ -1,5 +1,7 @@
 <script lang="ts">
-	import { createEventDispatcher } from 'svelte';
+	import { createEventDispatcher, onDestroy } from 'svelte';
+	import { basket, type Basket, type Item } from '$lib/stores/basket.js';
```
[the remark this answers is not in the document]
Comment (jcreek) [reply src/lib/components/SignOut.svelte]: Done

File: src/lib/components/SignOut.svelte [hunk 4]
```diff
@@ -9,10 +11,25 @@
...
+
+			// Clear the basket
+			localBasket = { items: [], subtotal: 0 };
+			basket.set(localBasket);
+			unsubscribe();
+			if (typeof localStorage !== 'undefined') {
+				localStorage.removeItem('basket');
+			}
 		});
 	}
```
[the remark this answers is not in the document]
Comment (jcreek) [reply src/lib/components/SignOut.svelte]: Done

**Comment by jcreek** [comment]
N.B. The build fails because I've added a reference to the logger, which isn't within this branch, but once it's merged into master it'll be fine :)

**Comment by jcreek** [comment]
@OllyNicholass there are NO coderabbit threads here that need your review, the rest have been actioned or had tickets created for them :)

**Comment by OllyNicholass** [comment]
image
It appears to be attempting to resolve the logger in the wrong location, and file format
image

**Comment by jcreek** [comment]
@OllyNicholass good spot! Fixed it :)

File: src/lib/components/SignOut.svelte [hunk 5]
```diff
@@ -9,11 +12,36 @@
...
+		}
+
+		try {
 			emitSignedOutEvent();
-		});
+
+			// Clear the basket
+			localBasket = { items: [], subtotal: 0 };
+			basket.set(localBasket);
+			unsubscribe();
```
[the remark this answers is not in the document]
Comment (jcreek) [reply src/lib/components/SignOut.svelte:36]: I don't believe this is actually an issue, but @OllyNicholass may disagree

File: src/lib/components/SignOut.svelte [hunk 6]
```diff
@@ -9,11 +11,36 @@
 	// Access the supabase client from the layout data
 	export let supabase: any;
```
[the remark this answers is not in the document]
Comment (jcreek) [reply src/lib/components/SignOut.svelte:14]: Done

**Comment by jcreek** [comment]
@OllyNicholass there are once again NO coderabbit threads here that need your review, the rest have been actioned or had tickets created for them :)

**Review by OllyNicholass — APPROVED** [review]

**Pull request merged**